_docs/src/documentation/rex_pipeline.md

```diff
@@ -105,40 +105,70 @@ It'll scan for all sharpmake files recursively in 3 different folders
-py _generate.py -sharpmake_args="/noClangTools" 
+# Default generation, but single threaded
+py _rex.py generate -sharpmake_args /multithreaded(false)
 ```
 
+### Code Generation
+The generation step also allows you to auto generate code. The settings for code generation for a particular project are located in the `config` folder and follow the same folder structure as their source root directory.
+
+Sharpmake supports enum and array generation.
+
+A code generation object always has a `key`. multiple projects can have an object with the same key and it's `content` will be merged together in the final source file.
+A key should have 1 and only 1 owner. An owner is a project's code generation object. This owner needs to specify the following
+- `Type` - What are we generating? can be `Enum` or `Array`
+- `Filepath` - This is the file where the code will be generated in
+
+for `Enums` the following needs to be specified
+- `ClassName` - This is the name of the enum class to be generated
+
+for `Arrays` the following needs to be specified
+- `Name` - This is the name of the variable that'll hold the content of the array
+- `ElementType` - This is the type the array will hold
+- `Includes` - This is the files that need to be included at the top of the file
+
 ## Building
-Although the previous explained generation step allows visual studio generation, the actual build pipeline of Rex uses ninja files as explained above.
+Building through Visual Studio and Visual Studio Code works the same as you would otherwise. We leverage the IDE's ability to run commandlines as build steps.
+These commandlines call into the actual build script which perform a build of a project in a specified configuration using a specified compiler.
+
+As mentioned above, the build script takes in the project, configuration and compiler you want to build with. It'll call the ninja files and their dependencies (if specified) and creates their output targets.
+
+After each build the `build_projects.json` file in the intermediate build folder is updated. This json file acts as a cache for the launch script so it knows what it can launch.
 
-Therefore the build script ignores the visual studio files and only cares about the ninja files, over which we have full control.
+### Build Types
+We have a few different kind of build types aka configurations, some are only supported with Clang as they are sanitizers.
+The different builds types are
 
-the script takes the project, compiler and configuration you want to compile for.
-After which it looks for this the ninja project reflecting the project you specified (which is auto generated by sharpmake) and compiles the configuration specified with the compiler you specified.
+- Debug: No optimizations enabled. All debug information is available.
+- Debug Opt: Optimized version of Debug build.
+- Release: All optimizations enabled, no debug information is available.
+- Coverage: Only enabled for Clang. Enables code coverage reports
```

Comment:
I would also explain code coverage and where those reports are located when generated.